src/main/java/org/candlepin/controller/PoolConverter.java

```diff
@@ -91,7 +90,15 @@ public Pool convertToPrimaryPool(SubscriptionInfo sub) {
             throw new IllegalStateException("Subscription references an incomplete product: " + pinfo);
         }
 
-        Product product = this.ownerProductCurator.getProductById(owner, pid);
+        // Resolve product ref using the global namespace
+        // TODO: FIXME: this may be incorrect, depending on the context of the caller. If this is
+        // called from an API method, this is incorrect and will lead to illegal/incorrect mappings.
+        // If this is called from the refresh/manifest flows, this is correct.
```

Comment:
Looking at the usage of this method, it is used in 2 places:
1) The Refresher
2) PoolRules, and looking at that call stack, it goes all the way to `public List<Pool> createAndEnrichPools(SubscriptionInfo sub)`, which is only used by unit tests, so it should be removed (along with all the intermediate methods).

Reply:
Also, it may be a good idea to group pool management classes by package, to indicate what/where they're supposed to belong. For example PoolConverter looks like it should only ever be used by Refresh/Import, so maybe it should go somewhere under `org.candlepin.controller.refresher` instead of just `org.candlepin.controller`

Reply:
I'll remove the `PoolRules` path, but I don't really want to think about reorganizing the classes I'm not working in at the moment. Should create a backlog task for that.

Reply:
After spending a couple hours poking at the test suite with the pool rules conversion removed, I'm opting to punt both to a separate task. The pool manager testing is an absolute mess. There are pool rules tests hanging out in `PoolManagerTest`, pool manager tests in `PoolCuratorTest`, and a bunch of unnecessary refresh loops for some reason. All of that would have been fine if PoolManagerTest wasn't an awful pile of clunky mocks which I really don't want to deal with in this already-XL sized PR.

As far as the line in question goes, I've updated it to use the namespace of the org from the subscription since it won't proceed without one anyway.